test/demo/utils/time.ts

```diff
@@ -0,0 +1,39 @@
+import {delay} from './delay';
+
```

Comment:
Is `demo/utils` the best place for this? the convention so far has been that everything under `demo` is running in an integration test environment, but this seems more like a unit testing util.

Reply:
I don't see anything wrong with having a `test/provider/utils` folder for unit testing utils, since we will be getting a lot more of them in the near future (hopefully)

Reply:
Hmm... It's annoying because I wanted to organise the unit tests to mirror the provider structuring - so the test for a thing in src/provider/foo/bar lives in test/provider/foo/bar. Then we're mixing the purpose of test/provider/utils - a mirror of src/provider/utils, or a folder of unit test utils.

Any thoughts on how best to solve this?

Reply:
Discussed offline. Have a structure that the two of us agree on.

Reply:
✅